knowledge-base/window-is-not-defined-nextjs.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Telerik React Report Viewer not working in NextJS
+description: Telerik React Report Viewer not working in NextJS
+type: troubleshooting
+page_title: Telerik React Report Viewer not working in NextJS
+slug: window-is-not-defined-nextjs
+position: 
+tags: Web, ReactJS, React, HTML5, Report Viewer
+ticketid: 1581204
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+		<tr>
+			<td>Project Type</td>
+			<td>NextJs</td>
+		</tr>
+		<tr>
+			<td>Report Viewer</td>
+			<td>React</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+The NextJS application cannot render the page with the `React Report Viewer`. Instead of the report viewer, the page shows a stack trace of errors.
+
+## Steps to Reproduce
+
+Create a new NextJS application through `create-next-app`, then add the [React Report Viewer]({%slug telerikreporting/using-reports-in-applications/display-reports-in-applications/web-application/react-report-viewer/react-report-viewer-overview%}) to a page and try to load that page.
+
+## Error Message
+
+````
+ReferenceError: window is not defined
+````
+
+## Cause
+
+`Next.js` is universal, which means it executes code first server-side, then client-side. The `window` object is only present client-side which is problem because our `React Report Viewer` interacts with it. 
+
+## Solution
+
+The solution is to render the component with the report viewer without `SSR`
+
+````JavaScript
+//create the dynamic component
+const DynamicReportViewer = dynamic(() => import('./ReportViewer').then(mod => mod.ReportViewer), {
+  ssr: false,
+})
+
+const Home: NextPage = () => {
+  return (
+    <div className={styles.container}>
+      <Head>
+        <title>Create Next App</title>
+        <meta name="description" content="Generated by create next app" />
+        <link rel="icon" href="/favicon.ico" />
+
+        <link href="https://kendo.cdn.telerik.com/2022.1.301/styles/kendo.common.min.css" rel="stylesheet" />
```

Comment:
Consider changing the Kendo version with the token from '_config.yaml', i.e. replace '2022.1.301' with '{{site.kendosubsetversion}}'. It will be automatically replaced during the generation of the documentation.